File: go-backend/internal/model/entity/plugin_config.go
package entity

import (
	"time"

	"gorm.io/plugin/soft_delete"
)

type PluginConfig struct {
	ID          int64                 `json:"id,string" gorm:"column:id;primaryKey;autoIncrement"`
	PluginKey   string                `json:"pluginKey" gorm:"column:pluginKey"`
	PluginName  string                `json:"pluginName" gorm:"column:pluginName"`
	PluginType  string                `json:"pluginType" gorm:"column:pluginType"`
	Description string                `json:"description" gorm:"column:description"`
	Config      string                `json:"config" gorm:"column:config"`
	Status      string                `json:"status" gorm:"column:status"`
	Priority    int                   `json:"priority" gorm:"column:priority"`
	CreateTime  time.Time             `json:"createTime" gorm:"column:createTime"`
	UpdateTime  time.Time             `json:"updateTime" gorm:"column:updateTime"`
	IsDelete    soft_delete.DeletedAt `json:"isDelete" gorm:"column:isDelete;softDelete:flag"`
}

func (PluginConfig) TableName() string {
	return "plugin_config"
}

File: python-backend/app/models/plugin_config.py
"""Plugin config model."""

from __future__ import annotations

from datetime import datetime

from sqlalchemy import BigInteger, DateTime, Index, Integer, String, Text, func, text
from sqlalchemy.dialects.mysql import TINYINT
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class PluginConfig(Base):
    __tablename__ = "plugin_config"
    __table_args__ = (
        Index("uk_pluginKey", "pluginKey", unique=True),
    )

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    plugin_key: Mapped[str] = mapped_column("pluginKey", String(64), nullable=False)
    plugin_name: Mapped[str] = mapped_column("pluginName", String(128), nullable=False)
    plugin_type: Mapped[str] = mapped_column(
        "pluginType",
        String(32),
        nullable=False,
        default="builtin",
        server_default=text("'builtin'"),
    )
    description: Mapped[str | None] = mapped_column(String(512), nullable=True)
    config: Mapped[str | None] = mapped_column(Text, nullable=True)
    status: Mapped[str] = mapped_column(
        String(32),
        nullable=False,
        default="active",
        server_default=text("'active'"),
    )
    priority: Mapped[int] = mapped_column(Integer, nullable=False, default=100, server_default=text("100"))
    create_time: Mapped[datetime] = mapped_column(
        "createTime",
        DateTime,
        nullable=False,
        default=datetime.utcnow,
        server_default=func.current_timestamp(),
    )
    update_time: Mapped[datetime] = mapped_column(
        "updateTime",
        DateTime,
        nullable=False,
        default=datetime.utcnow,
        onupdate=datetime.utcnow,
        server_default=func.current_timestamp(),
    )
    is_delete: Mapped[int] = mapped_column("isDelete", TINYINT, nullable=False, default=0, server_default=text("0"))
